src/shims/x86/sse41.rs

```diff
@@ -0,0 +1,312 @@
+                let res = op0.round_to_integral(rounding).value;
+                this.write_scalar(Scalar::from_f32(res), &this.project_index(&dest, 0)?)?;
+
+                for i in 1..dest_len {
+                    this.copy_op(
+                        &this.project_index(&left, i)?,
+                        &this.project_index(&dest, i)?,
+                        /*allow_transmute*/ false,
+                    )?;
+                }
+            }
+            // Used to implement the _mm_floor_sd, _mm_ceil_sd and _mm_round_sd
+            // functions. Rounds the first element of `right` according to `rounding`
+            // and copies the remaining from `left`.
+            "round.sd" => {
+                let [left, right, rounding] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+
+                let (left, left_len) = this.operand_to_simd(left)?;
+                let (right, right_len) = this.operand_to_simd(right)?;
+                let (dest, dest_len) = this.place_to_simd(dest)?;
+
+                assert_eq!(dest_len, left_len);
+                assert_eq!(dest_len, right_len);
+
+                let rounding = match this.read_scalar(rounding)?.to_i32()? {
+                    0x00 | 0x80 => rustc_apfloat::Round::NearestTiesToEven,
+                    0x01 | 0x81 => rustc_apfloat::Round::TowardNegative,
+                    0x02 | 0x82 => rustc_apfloat::Round::TowardPositive,
+                    0x03 | 0x83 => rustc_apfloat::Round::TowardZero,
+                    rounding => throw_unsup_format!("unsupported rounding mode 0x{rounding:02x}"),
+                };
+
+                let op0 = this.read_scalar(&this.project_index(&right, 0)?)?.to_f64()?;
+                let res = op0.round_to_integral(rounding).value;
+                this.write_scalar(Scalar::from_f64(res), &this.project_index(&dest, 0)?)?;
+
+                for i in 1..dest_len {
+                    this.copy_op(
+                        &this.project_index(&left, i)?,
+                        &this.project_index(&dest, i)?,
+                        /*allow_transmute*/ false,
+                    )?;
+                }
+            }
+            // Used to implement the _mm_minpos_epu16 function.
+            // Find the minimum unsinged 16-bit integer in `op` and
+            // returns its value and position.
+            "phminposuw" => {
+                let [op] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+
+                let (op, op_len) = this.operand_to_simd(op)?;
+                let (dest, dest_len) = this.place_to_simd(dest)?;
+
+                // Find minimum
+                let mut min_value = u16::MAX;
+                let mut min_index = 0;
+                for i in 0..op_len {
+                    let op = this.read_scalar(&this.project_index(&op, i)?)?.to_u16()?;
+                    if op < min_value {
```

Comment:
Do we have a test that distinguishes `<` from `<=` here? That obviously returns the same value, but not the same index.

Reply:
There will be.